app/components/chat_message/chat_message.html.erb

```diff
@@ -9,13 +9,15 @@
   data-chat-message-copy-url-value="<%= request_url(message.request, anchor: id) %>"
   id="<%= id %>"
 >
-    <% if message.text.present? %>
-      <header class="ChatMessage-header">
-        <% if message.sender %>
-          <%= c 'avatar', record: message.sender, style: :small, class: "ChatMessage-avatar" %>
-        <% else %>
-          <%= c 'avatar', style: :small, class: "ChatMessage-avatar" %>
-        <% end %>
+    <header class="ChatMessage-header">
+      <% sent_by_x_at = t('.sent_by_x_at', name: message.sender ? message.sender.first_name : Setting.project_name, date: date_time(message.updated_at)).html_safe %>
```

Comment:
In general I like the idea but IMHO opinion this only works partially:
- For the request view it's fine to remove the repeated request as it is redundant information
- On the chronological view though - you need the date & time and a reference to the request context
So the only thing we can do really is to omit the request title link for the request view. I decided not to include this due to parameter drilling ( played around with that switch_display) for an in my opinion minor improvement. A kind of hacky alternative would be just to display none for some elements based on the CSS hierarchy if we agree this is only display nicety.

Reply:
see a118eb1cb09fdeaa2255252abcd669229e84cd28 for a CSS proposal

Reply:
see 3573e2862358ed1e79205bad65cab998835b1622 for a refactoring of the view logic

Reply:
I liked both your changes. It would be an improvement, in my opinion, if we could display the same information on the conversations page chat message components, but have text instead of a link to its own page. We could achieve this with:

```rb
if message.sent_from_contributor?
  link_to_unless(current_page?(check_parameters: false), sent_by_x_at, conversations_contributor_path(id: message.contributor.id, anchor: id), data: { turbo: false })
else
  content_tag(:p, sent_by_x_at)
end
```

Reply:
as you insist so much on this so much..

Reply:
I'm sorry if I came across as insisting on this change as that was not my intention. I brought up a concern, which admittedly is not that important, and a possible solution. You only partially implemented a solution that address the concern and explained why you didn't want to implement the other part. I saw that your reason was valid and suggested another way to solve it that addresses the concern while adding some complexity. I did not request changes in the PR review specifically because I did not want to insist on changes.

If you don't feel the change I suggest is worth the complexity, I'm happy for you to simply say that you will not implement it and we move on, or we have a follow-up discussion on when you think adding complexity is worth it and when I think it's worth it. The last part simply for getting to know each other since we have never worked on a project before.